src/lvm.cpp

```diff
@@ -0,0 +1,451 @@
+            "LVM ERROR: Failed to check if LVM is enabled.");
+    }
+}
+
+void LVM::checkThinProvisioning()
+{
+    std::list<std::string> output;
+    const std::string checkThinCommand = "lvs -o+lv_layout --noheadings";
+    int exitCode = cloyster::runCommand(checkThinCommand, output, false);
+
+    if (exitCode == 0) {
+        auto it = std::find_if(
+            output.begin(), output.end(), [](const std::string& line) {
+                return line.find("thin") != std::string::npos;
+            });
+
+        if (it != output.end()) {
+            LOG_TRACE("LVM Thin provisioning is enabled.\n");
+        } else {
+            LOG_WARN("LVM Thin provisioning is disabled.");
+        }
+    } else {
+        LOG_WARN("Failed to check thin provisioning status.");
+    }
+}
+
+void LVM::checkEnoughDiskSpaceAvailable()
+{
+    std::list<std::string> output;
+    const std::string checkDiskSpaceCommand
+        = "vgs --noheadings -o vg_name,vg_size,vg_free --units G";
+
+    int exitCode = cloyster::runCommand(checkDiskSpaceCommand, output, false);
+
+    if (exitCode == 0 && !output.empty()) {
+        try {
+            bool allVGsHaveEnoughSpace = true;
+
+            for (const std::string& line : output) {
+                LOG_DEBUG("LVM: Check disks space - Raw output line: {}", line);
+
+                // Remove leading/trailing whitespaces
+                std::string cleanLine = line;
+                cleanLine.erase(cleanLine.begin(),
+                    std::find_if(cleanLine.begin(), cleanLine.end(),
+                        [](unsigned char ch) { return !std::isspace(ch); }));
+                cleanLine.erase(
+                    std::find_if(cleanLine.rbegin(), cleanLine.rend(),
+                        [](unsigned char ch) { return !std::isspace(ch); })
+                        .base(),
+                    cleanLine.end());
+
+                if (cleanLine.empty()) {
+                    LOG_DEBUG("LVM: Skipping empty or malformed line.");
+                    continue;
+                }
+
+                // Split the line into vg_name, vg_size, and vg_free
+                std::istringstream iss(cleanLine);
+                std::string vgName, vgSizeStr, vgFreeStr;
+                if (!(iss >> vgName >> vgSizeStr >> vgFreeStr)) {
+                    LOG_DEBUG("LVM: Failed to split line into expected fields. "
+                              "Line: {}",
+                        cleanLine);
+                    throw std::runtime_error(
+                        "LVM ERROR: Failed to parse volume group information.");
+                }
+
+                // Remove 'G' from size and free space strings
+                vgSizeStr.erase(
+                    std::remove(vgSizeStr.begin(), vgSizeStr.end(), 'G'),
+                    vgSizeStr.end());
+                vgFreeStr.erase(
+                    std::remove(vgFreeStr.begin(), vgFreeStr.end(), 'G'),
+                    vgFreeStr.end());
```

Comment:
Should not do that. It's the unit. What if it's terabytes instead of gigabytes?

Reply:
Can't receive in terabyte format because I'm calling  `vgs` with `--units G`